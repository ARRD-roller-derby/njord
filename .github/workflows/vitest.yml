name: Vitest
on: push
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v2
      with:
        node-version: '16.x'
    - name: Install dependencies
      run: npm ci
    - name: Run Vitest tests
      run: npm run coverage
    - uses: actions/upload-artifact@v2
      if: always()
      with:
        name: vitest-report
        path: vitest-report/
        retention-days: 30