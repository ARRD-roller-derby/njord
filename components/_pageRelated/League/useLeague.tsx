import { useState } from 'react';

export default function useLeague(){
  const [canIRequest,setCanIRequest] = useState<boolean>(true);
  return {canIRequest,setCanIRequest}
}