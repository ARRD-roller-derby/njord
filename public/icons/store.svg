<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg
   xmlns:dc="http://purl.org/dc/elements/1.1/"
   xmlns:cc="http://creativecommons.org/ns#"
   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
   xmlns:svg="http://www.w3.org/2000/svg"
   xmlns="http://www.w3.org/2000/svg"
   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
   viewBox="0 0 576 512"
   version="1.1"
   id="svg10"
   sodipodi:docname="store-duotone.svg"
   inkscape:version="1.0.1 (3bc2e813f5, 2020-09-07)">
  <metadata
     id="metadata14">
    <rdf:RDF>
      <cc:Work
         rdf:about="">
        <dc:format>image/svg+xml</dc:format>
        <dc:type
           rdf:resource="http://purl.org/dc/dcmitype/StillImage" />
      </cc:Work>
    </rdf:RDF>
  </metadata>
  <sodipodi:namedview
     pagecolor="#ffffff"
     bordercolor="#666666"
     borderopacity="1"
     objecttolerance="10"
     gridtolerance="10"
     guidetolerance="10"
     inkscape:pageopacity="0"
     inkscape:pageshadow="2"
     inkscape:window-width="1920"
     inkscape:window-height="1009"
     id="namedview12"
     showgrid="false"
     inkscape:zoom="0.83984375"
     inkscape:cx="329.78442"
     inkscape:cy="254.52185"
     inkscape:window-x="1272"
     inkscape:window-y="-8"
     inkscape:window-maximized="1"
     inkscape:current-layer="svg10" />
  <!--! Font Awesome Pro 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) Copyright 2022 Fonticons, Inc. -->
  <defs
     id="defs4">
    <style
       id="style2">.fa-secondary{opacity:.4}</style>
  </defs>
  <path
     class="fa-primary"
     d="M490.3 13.13L547.6 103.8C577.3 150.7 551 215.8 495.5 223.2C491.6 223.7 487.6 224 483.4 224C457.4 224 434.2 212.6 418.3 195C402.4 212.6 379.2 224 353.1 224C327 224 303.8 212.6 287.9 195C272 212.6 248.9 224 222.7 224C196.7 224 173.5 212.6 157.6 195C141.7 212.6 118.5 224 92.36 224C88.3 224 84.21 223.7 80.24 223.2C24.92 215.8-1.255 150.6 28.33 103.8L85.66 13.13C90.76 4.979 99.87 0 109.6 0H466.4C476.1 0 485.2 4.978 490.3 13.13H490.3z"
     id="path6"
     style="fill:#999999" />
  <path
     class="fa-secondary"
     d="M80.23 223.2C84.21 223.7 88.3 224 92.36 224C105.1 224 117.2 221.3 128 216.5V384H448V216.5C458.8 221.3 470.8 224 483.4 224C487.6 224 491.6 223.7 495.5 223.2C501.4 222.4 506.9 221 512 219.1V448C512 483.3 483.3 512 448 512H128C92.65 512 64 483.3 64 448V219.1C69.08 221 74.5 222.4 80.24 223.2H80.23z"
     id="path8"
     style="fill:#b3b3b3" />
</svg>
