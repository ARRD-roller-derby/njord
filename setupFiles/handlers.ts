import { rest } from 'msw'

export const handlers = [
  rest.get('/api/auth/session', (_req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json({
        user: {
          _id: 'idtest',
          email: 'test@test.fr',
          emailVerified: '2022-08-04T22:31:56.765Z',
          profiles: ['admin'],
          allergies: [],
          name: 'Benoist',
          lastname: 'Bouteiller',
          derbyName: 'Enrico Motion',
          numRoster: '95',
          admin: true,
          teams: [],
          wallet: 200,
        },
        expires: '2022-09-04T13:44:07.190Z',
        isAdmin: true,
      })
    )
  }),
  rest.post('/api/notifications/notifications', (_req, res, ctx) => {
    return res(ctx.status(200), ctx.json([]))
  }),
  rest.post('/api/notifications/count', (_req, res, ctx) => {
    return res(ctx.status(200), ctx.text('10'))
  }),
  rest.post('/api/requests/requests', (_req, res, ctx) => {
    return res(ctx.status(200), ctx.json([]))
  }),
  rest.post('/api/users/wallet', (_req, res, ctx) => {
    return res(ctx.status(200), ctx.text('200'))
  }),
  rest.post('/api/users/avatar', (_req, res, ctx) => {
    return res(ctx.status(200), ctx.json('/public/static/images/profile.webp'))
  }),
  rest.post('/api/league/users', (_req, res, ctx) => {
    return res(ctx.status(200), ctx.json([]))
  }),
  rest.post('/api/users/me', (_req, res, ctx) => {
    return res(ctx.status(200), ctx.json({
      email: 'email@email.email',
      name:'name',
      lastname:'lastname'
    }))
  }),
]
