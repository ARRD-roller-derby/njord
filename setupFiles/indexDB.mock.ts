import "fake-indexeddb/auto";

//Mock indexDb API