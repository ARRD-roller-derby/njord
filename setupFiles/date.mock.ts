import { vi} from 'vitest'
import { registerLocale, setDefaultLocale } from 'react-datepicker'
import frFns from 'date-fns/locale/fr'

registerLocale('fr', frFns)
setDefaultLocale('fr')

const date = new Date(2022, 7, 7,7)
vi.setSystemTime(date)
