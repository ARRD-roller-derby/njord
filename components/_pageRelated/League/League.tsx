import AuthentificatedLayout from '../../_layouts/Authentificated/Authentificated';
import PendingRequestIndicator from '../../PendingRequestIndicator/PendingRequestIndicator';
import MyLeague from '../../League/MyLeague/MyLeague';
import useLeague from './useLeague';

export default function League(){
  const {canIRequest,setCanIRequest} = useLeague();

  return <AuthentificatedLayout>
    <PendingRequestIndicator setCanIRequest={setCanIRequest} />
    <MyLeague canIRequest={canIRequest}/>
  </AuthentificatedLayout>
}