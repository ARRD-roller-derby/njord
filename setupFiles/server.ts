
import { setupServer } from 'msw/node'
import { handlers } from './handlers'

export const server = setupServer(...handlers)

beforeAll(() => {
  // Enable the mocking in tests.
  server.listen({ onUnhandledRequest: 'warn'})
})

afterEach(() => {
  // Reset any runtime handlers tests may use.
  server.resetHandlers()
})

afterAll(() => {
  // Clean up once the tests are done.
  server.close()
})