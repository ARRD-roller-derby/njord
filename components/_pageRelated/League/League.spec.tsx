import { afterEach,describe,expect, it, vi } from 'vitest'
import { cleanup } from '@testing-library/react'
import League from './League'
import { render, screen } from '../../../utils/test-utils'
import { rest } from 'msw'
import { server } from '../../../setupFiles/server'
import { SocketContext } from '../../../stores/socket.store'

describe('<League />', () => {
  afterEach(cleanup)
  server.use(
    rest.post('/api/requests/league/pending', (_req, res, ctx) => {
      return res(ctx.status(200), ctx.text('Vous avez une demande en attente'))
    }),
    rest.post('/api/leagues/leagues', (_req, res, ctx) => {
      return res(ctx.status(200), ctx.json([]))
    })
  );

  it('Check snapshot', async () => {
    const {asFragment } = render(
      <SocketContext.Provider value={[null, vi.fn]}>
        <League/>
      </SocketContext.Provider>
    )

    expect(await screen.findByText('200 Dr.')).toBeInTheDocument()
    expect(await screen.findByText("Vous n'avez pas de league")).toBeInTheDocument()
    
    expect(asFragment()).toMatchSnapshot()
    
  })
})
