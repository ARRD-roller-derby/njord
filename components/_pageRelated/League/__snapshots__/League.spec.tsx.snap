// Vitest Snapshot v1

exports[`<League /> > Check snapshot 1`] = `
<DocumentFragment>
  <div
    class="_container_lmcue_1"
    data-color-mode="dark"
  >
    <header
      class="_header_lmcue_18"
    >
      <header
        class="_header_1yhnh_1"
      >
        <div
          class="_appName_1yhnh_7"
        >
          <a
            class="_title_1045q_1"
            title="retour à l'accueil"
          >
            Njörd
          </a>
          <div
            class="_link_1ld19_1"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="flêche"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <div>
              Njörd
            </div>
          </div>
        </div>
        <div
          class="_appName_1yhnh_7"
        >
          <div
            class="_count_snjyq_1"
          >
            200 Dr.
          </div>
          <a
            class="_container_1ssp7_1"
            data-active="false"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="clôche"
                class="_bell_1ssp7_11"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <div
              class="_count_13krl_1"
            >
               10
            </div>
          </a>
          <div
            class="_container_1dfbn_1"
          >
            <div
              class="_avatar_1dfbn_12"
              data-ignore="true"
              data-testid="avatar"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2730%27%20height=%2730%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="/public/static/images/profile.webp"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
          </div>
        </div>
      </header>
      <div
        class="Toastify"
      />
    </header>
    <div
      class="_menu_lmcue_22"
    />
    <main
      class="_main_lmcue_12"
    >
      <div
        class="_box_lmcue_31"
      >
        <div
          class="_container_1diov_1"
        >
          <div
            class="_alert_1diov_9"
          >
            Vous n'avez pas de league
          </div>
          <button>
            Rejoindre une league
          </button>
          <div
            class="_users_1diov_32"
          >
            <div
              class="_container_54422_1"
            >
              <div
                class="_search_54422_16"
              >
                <div
                  class="_container_1hxwo_1"
                >
                  <input
                    class="_input_1hxwo_55"
                    placeholder="Rechercher"
                    value=""
                  />
                  <div
                    class="_cross_1hxwo_18"
                    data-disabled="true"
                  />
                </div>
              </div>
              <div
                class="_container_54422_1"
              >
                <div
                  class="_box_54422_6"
                >
                  <div
                    class="_users_54422_26"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
